extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/DefaultSerdeDiscoveryState.java

```diff
@@ -0,0 +1,143 @@
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.Type;
+
+import io.smallrye.reactive.messaging.kafka.KafkaConnector;
+
+class DefaultSerdeDiscoveryState {
+    private final IndexView index;
+
+    private final Map<String, Boolean> isKafkaConnector = new HashMap<>();
+
+    private Boolean hasConfluent;
+    private Boolean hasApicurio1;
+    private Boolean hasApicurio2;
+    private Boolean hasJackson;
+    private Boolean hasJsonb;
+
+    DefaultSerdeDiscoveryState(IndexView index) {
+        this.index = index;
+    }
+
+    boolean isKafkaConnector(boolean incoming, String channelName) {
+        String channelType = incoming ? "incoming" : "outgoing";
+        return isKafkaConnector.computeIfAbsent(channelType + "|" + channelName, ignored -> {
+            String connectorKey = "mp.messaging." + channelType + "." + channelName + ".connector";
+            String connector = ConfigProvider.getConfig()
+                    .getOptionalValue(connectorKey, String.class)
+                    .orElse("ignored");
+            return KafkaConnector.CONNECTOR_NAME.equals(connector);
+        });
+    }
+
+    boolean isAvroGenerated(DotName className) {
+        ClassInfo clazz = index.getClassByName(className);
+        return clazz != null && clazz.classAnnotation(DotNames.AVRO_GENERATED) != null;
+    }
+
+    boolean hasConfluent() {
+        if (hasConfluent == null) {
+            try {
+                Class.forName("io.confluent.kafka.serializers.KafkaAvroDeserializer", false,
+                        Thread.currentThread().getContextClassLoader());
+                hasConfluent = true;
+            } catch (ClassNotFoundException e) {
+                hasConfluent = false;
+            }
+        }
+
+        return hasConfluent;
+    }
+
+    boolean hasApicurio1() {
+        if (hasApicurio1 == null) {
+            try {
+                Class.forName("io.apicurio.registry.utils.serde.AvroKafkaDeserializer", false,
+                        Thread.currentThread().getContextClassLoader());
+                hasApicurio1 = true;
+            } catch (ClassNotFoundException e) {
+                hasApicurio1 = false;
+            }
+        }
+
+        return hasApicurio1;
+    }
+
+    boolean hasApicurio2() {
+        if (hasApicurio2 == null) {
+            try {
+                Class.forName("io.apicurio.registry.serde.avro.AvroKafkaDeserializer", false,
+                        Thread.currentThread().getContextClassLoader());
+                hasApicurio2 = true;
+            } catch (ClassNotFoundException e) {
+                hasApicurio2 = false;
+            }
+        }
+
+        return hasApicurio2;
+    }
+
+    boolean hasJackson() {
+        if (hasJackson == null) {
+            try {
+                Class.forName("com.fasterxml.jackson.databind.ObjectMapper", false,
```

Comment:
Something I fixed yesterday. Kafka now unconditionally depends on Jackson.

Reply:
Hmm, does it have to, or is it a convenience?

Reply:
It has to. Kafka is now using Jackson to handle authentication, and other stuff too.

Reply:
Interesting. Alright, I'll simplify this.